core/io/image.h

```diff
@@ -333,13 +342,13 @@ class Image : public Resource {
 
 	Error load(const String &p_path);
 	static Ref<Image> load_from_file(const String &p_path);
-	Error save_png(const String &p_path) const;
-	Error save_jpg(const String &p_path, float p_quality = 0.75) const;
+	Error save_png(const String &p_path, int p_integrity_level = 0) const;
```

Comment:
Same here, why not `FileAccess::SaveIntegrityLevel`?

Reply:
The only reason I'm using `int` instead of `FileAccess::SaveIntegrityLevel` is to avoid another `#include` as I need to bring file_access.h in. I've seen that pattern in other parts of the API, but used inconsistently.

Personally, I prefer strong type safety; and the fact that header inclusion is exploding throughout the database should be addressed separately(\*).

I can easily change it to `FileAccess::SaveIntegrityLevel`.
Thoughts?


(\*) Something needs to be done about that. A single header is changed and between 25-75% of the project needs to be built again. It's really bad for productivity.

Reply:
For unexposed `ResourceFormatSaver` it's OK as is.

But I would change it for public `Image` methods, some parts of the code avoid includes from other parts of the complex dependencies, but both `Image` and `FileAccess` are part `core/io`, so adding extra include should be fine.

Reply:
`ResourceFormatSaver` is part of `core/io` too, so if that's the criteria we can add the header there too.

On the other hand, ResourceFormatSaver is also included in a lot of other modules headers, while file_access.h isn't.

Reply:
I stand corrected. Image is included in a lot of other module headers too. It makes sense for `ResourceFormatSaver` to also become typed.